src/lib/commandline_cmds.ts

```diff
@@ -155,6 +155,42 @@ export function getCommandlineFns(cmdline_state) {
             return messageOwnTab("controller_content", "acceptExCmd", [command])
         },
 
+        execute_ex_on_completion: (excmd: string) => {
+            const command = cmdline_state.getCompletion()
+
+            cmdline_state.fns.hide_and_clear()
```

Comment:
So you're saying even without this line the command line hides?

Reply:
Interestingly, without line 161 `cmdline_state.fns.hide_and_clear()`,  the command line hides in the case of `<C-o>yy` but not in the case of `<S-Del>`.  Hiding the command line upon ` <C-o>yy` is consistent with the previous behavior.

Another consideration: even when the cmdline persists after deleting a tab from the "Tabs" or "Tabs All" buffer with `<S-Del>`, the deleted tab is not dynamically removed from the tab list.  To see the updated list of tabs, I exit the cmd mode and press 'b' again.  I'm working on fixing this, so that the tab buffer dynamically updates the tab list if a tab is deleted.

Reply:
There's two possible causes for that:

- we're using `clipboard yank` rather than `yank` which gives feedback via the command line
- we use the command line to copy text - see #2597 

For the update, you could either:
- make composite work with `ex.` so we can call refresh_completions (or whatever it is called) or
- add a listener that updates the completions as soon as the tabs change

Reply:
The listener is a little tricky now that I think about it, as the events are fired in the background script but our completions run in the content script.

I think I'd recommend the following:

1. For each of the tab events we care about here, https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs#Events in src/background.ts register a listener with them
2. That listener should send a message to all the tabs telling them that the tabs have been updated - I think we have the `messageAllTabs` function for this
3. Listen for that message in src/completions/Tab{,All}.ts and update accordingly. For a first pass I'd ignore the content of the message and just re-run the normal update process whenever a message about a tabs update is received; we can optimise later if we need to.

---

That is slightly involved, though, so it might be better to merge this as-is and make a new issue for that. What do you think?

Reply:
Sounds good to make a new issue to add the tab event listener.

I did add another commit to move the duplicate logic that checks if we should store the ex string command in the command history to its own function.